submissions/crealime/oop-exercise/index.js

```diff
@@ -0,0 +1,197 @@
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/crealime/a-tiny-JS-world
+   Web app: https://crealime.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+
+const pronounsHisHer = {
+  male: 'His',
+  female: 'Her'
+}
+
+const pronounsHeShe = {
+  male: 'He',
+  female: 'She'
+}
+
+function getRandomId() {
+  return  new Date().getTime() + '_' + Math.floor(10000 + Math.random() * (99999 + 1 - 10000))
+}
+
+class Inhabitant {
+  constructor(species, name, gender, legs, saying) {
+    this.id = getRandomId()
+    this.species = species
+    this.name = name
+    this.gender = gender
+    this.legs = legs
+    this.saying = saying
+    this.friends = []
+  }
+
+  addFriends(...newFriends) {
+    this.friends = [...this.friends, ...newFriends]
+  }
+
+  getAcquaintanceInfo() {
+    return `This is ${this.name} and ${pronounsHeShe[this.gender].toLowerCase()} is a ${this.species}.`
+  }
+
+  getGenderInfo() {
+    return `${pronounsHisHer[this.gender]} gender is ${this.gender}.`
+  }
+
+  getLimbsInfo() {
+    return `${pronounsHeShe[this.gender]} have ${this.legs || 0} ${this.legs === 1 ? 'leg' : 'legs'}.`
+  }
+
+  getFriendsInfo() {
+    return `${this.name} have ${this.friends.length} ${this.friends.length === 1 ? 'friend' : 'friends'}${this.friends.length > 0 ? ': ' + this.friends.map(friend => friend.name).join(', ') : ''}.`
+  }
+
+  getSayingInfo() {
+    return `${pronounsHeShe[this.gender]} usually says hello like this: ${this.saying}`
+  }
+
+  getFullInfo() {
+    return `${this.getAcquaintanceInfo()} ${this.getGenderInfo()} ${this.getLimbsInfo()} ${this.getFriendsInfo()} ${this.getSayingInfo()}`
+  }
+}
+
+class Dog extends Inhabitant {
+  constructor(name, gender, legs) {
+    super('dog', name, gender, legs, 'Woof-woof!')
+  }
+}
+
+class Cat extends Inhabitant {
+  constructor(name, gender, legs) {
+    super('cat', name, gender, legs, Cat.say())
+  }
+
+  static say() {
+    return 'Meow-meow!'
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(name, surname, gender, legs, hands, saying) {
+    super('human', name, gender, legs, saying)
+    this.surname = surname
+    this.hands = hands
+  }
+
+  getAcquaintanceInfo() {
+    return super.getAcquaintanceInfo().replace(this.name, this.name + ' ' + this.surname)
+  }
+
+  getLimbsInfo() {
+    return `${pronounsHeShe[this.gender]} have ${this.hands || 0} ${this.hands === 1 ? 'hand' : 'hands'} and ${this.legs || 0} ${this.legs === 1 ? 'leg' : 'legs'}.`
+  }
+
+  sayFullInfo() {
+    return `${this.saying} My name is ${this.name} ${this.surname} and I am a ${this.species}. My gender is ${this.gender}. I have ${this.hands || 0} ${this.hands === 1 ? 'hand' : 'hands'} and ${this.legs || 0} ${this.legs === 1 ? 'leg' : 'legs'}. I have ${this.friends.length} ${this.friends.length === 1 ? 'friend' : 'friends'}${this.friends.length > 0 ? ': ' + this.friends.map(friend => friend.name).join(', ') : ''}.`
```

Comment:
Looks quite heavy to read. As a peer developer I would refrain to take a task that would require to introduce changes to this code (adding a property, changing property handling).
Code is written for other people. Take care of others. You will want other developers take care of you.

You may want to introduce a method that would return a plural form of word along with its value. This would make the code shorter and more readable. Not sure if it helped to make it more maintainable and scalable.